### ISSUE TITLE
Dynamic measurement is still not working

### ISSUE THREAD

**Issue by ekawatani** [tannerlinsley/react-virtual#235]
Hi,

I spent hours on this, but I still don't believe dynamic measurement is really working in react-virtual. FYI, I read this issue already: #23 

This is the example from the doc, but it has hardcoded width and height based on `${virtualColumn.size}px`, and `${virtualRow.size}px`. The _resulting_ dimensions are controlled by these values and they are not dynamic in this example.
https://codesandbox.io/s/github/tannerlinsley/react-virtual/tree/master/examples/dynamic

In my case, I have fixed column widths and every cell can have dynamic height. I don't think this is supported. Here are my observations.

1. I tried removing the hardcoded width and height from the example, but as soon as I do so, I get an error that "Maximum update depth exceeded". It seems it could not calculate the dimensions properly and got stuck in a loop. (Is this a bug?)

2. Removing only the height from the linked example above makes each cell to have the same height. However, this only works only if _every cell in the same row has the same height._ As soon as I make several cells have different height within the same row, I get the "Maximum update depth exceeded" error again.

It's dynamic in a sense that each _row_ can have different heights, but not each _cell_. (And, this is only for using fixed column width. I'm not sure how this would even work if both row and column had dynamic size.) I wouldn't call this behavior dynamic. 

If it works if every cell in the same row has the same height, wouldn't it possible to take the maximum height in each row use it across all cells in the row? I'm not sure why it crashes with the error.

**Comment by piecyk** [comment]
Hi, yes examples are wrong in dynamic context, width and height should not be set, we need to update them soonish. If you could provide [codesanbox](https://codesandbox.io/) then we can check what's wrong. Also have a look at https://github.com/tannerlinsley/react-virtual/issues/28 there i explain bit more caveats with measureRef.

**Comment by ekawatani** [comment]
Thanks!

It'll crash as soon as you remove these two lines.
![image](https://user-images.githubusercontent.com/61330990/147618232-07fd6c30-1fae-42c5-aeb1-318f5689e24a.png)

Not sure if it's helpful, but I created a sandbox [here](https://codesandbox.io/s/summer-lake-zce89).

**Comment by ekawatani** [comment]
And here's another one that I explained in 2 in the original post. Cells in each row have different heights in this one.

https://codesandbox.io/s/fervent-glitter-2v6s9

**Comment by piecyk** [comment]
> It'll crash as soon as you remove these two lines. This is what I explained in "1" in my original post.

Ooo, it throws an error because the width is different in some rows for same columns, basic column width must be the exact for each row, same for row.

> And here's another one that I explained in "2" in the original post. Cells in each row have different heights in this one. I removed the hardcoded height, but kept the hardcoded width.

Same story, i think you are after masonry layout, so we would need to have virtualizer for each row/column? 🤔 

fyi created pr with updated examples https://github.com/tannerlinsley/react-virtual/pull/236

**Comment by ekawatani** [comment]
Thanks, I'd expect it to pick the tallest/widest cell and other cells would inherit the same size. This would mean we'll need to scan across every visible cell and that's possibly expensive to do, but this could be something nice to add to the documentation (Just a suggestion ;)).

By the way, I'm not quite sure what you updated in the PR. I visited the link, but the example still has hardcoded width and height.

**Comment by piecyk** [comment]
> Thanks, I'd expect it to pick the tallest/widest cell and other cells would inherit the same size. 

Sound like good example to add 😉 

> By the way, I'm not quite sure what you updated in the PR. I visited the link, but the example still has hardcoded width and height.

The trick here is to move hardcoded values to child element, not where we attach the measureRef, this how we simulate the dynamic size.

**Comment by ekawatani** [comment]
Thanks, I'll leave it to you whether you want to close this issue as the duplicate of the other issue you mentioned. I wanted to know whether this was supported, and I moved on to `react-virtualized` for now which seems to support some of this.

**Comment by tannerlinsley** [comment]
We do need to get the examples fixed, but other than that, the dynamic sizing should work just fine. As for the masonry layout features, we'll likely need to work our way into that in the next major version. Closing for now.

**Comment by lukesmurray** [comment]
I just ran into this so I'll add the solution that worked for me. Try setting a maximum width/height on your virtualized elements. If you have virtualized columns set a maximum width and if you have virtualized rows set a maximum height. In my case, I was virtualizing columns and one of the columns was growing to fill the entire available width. When I added the maximum width constraint, the update loop went away, and I was able to diagnose the problem.
